### ISSUE TITLE
RWX/CephFS mount fails randomly (attach succeeds)

### ISSUE THREAD

**Issue by guimou** [rook/rook#5766]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
Some containers (part of Argo workflows) sometimes fail to mount RWX/CephFS shared volumes. The volume attach always succeeds, but sometimes the mount fails and the container creation times out. It's not totally reproducible, sometimes it works, sometimes not...
Other info: 
- 6 different volumes are mounted simultaneously in each container, and multiple containers are launched at the same time (cron jobs to gather and store data).
- We have a staging project, on the same OCP cluster, with the exact same configuration/images/storage/..., and everything is working flawlessly every time.

**Expected behavior:**
Mounts should happen without problem.

**File(s) to submit**:
Complete csi-cephfsplugin logs (extracted with Stern, so format is PodName ContainerName Log)

**Environment**:
* Rook version: 1.3.5
* Storage backend version: Ceph 14.2.9
* Kubernetes version: 1.16.2 
* Kubernetes cluster type: OpenShift 4.3.18
* Storage backend status: HEALTH_OK

**Comment by guimou** [comment]
Adding logs
[csi-cephfs-logs.zip](https://github.com/rook/rook/files/4880462/csi-cephfs-logs.zip)

**Comment by travisn** [comment]
@joscollin Could you take a look at this to investigate why the cephfs mounts sometimes fail?

**Comment by guimou** [comment]
Some further information as we were able to reproduce the issue.
- Some of the pods succeeded. But all those which failed had been scheduled on the same node.
![Screenshot_20200706_145252](https://user-images.githubusercontent.com/3944034/86630179-7ced0b80-bf9a-11ea-863e-c0e989fe1341.png)
- In the CSI plugin of the corresponding node there were those kind of errors:
```
E0704 03:02:01.901168       1 utils.go:163] ID: 45514 Req-ID: 0001-0009-rook-ceph-0000000000000001-cba856b8-bd42-11ea-91b6-0a580a800279 GRPC error: rpc error: code = Aborted desc = an operation with the given Volume ID 0001-0009-rook-ceph-0000000000000001-cba856b8-bd42-11ea-91b6-0a580a800279 already exists
```
I'm attaching the full log.
[csi-cephfsplugin-f94sw-csi-cephfsplugin.log](https://github.com/rook/rook/files/4880815/csi-cephfsplugin-f94sw-csi-cephfsplugin.log)
- I checked all the other CSI pods, some of them have the same king of errors, some don't.
- As suggested by someone, I checked the dmesg on the host itself. Nothing specific for the test period, only this (mon communications):
![Screenshot_20200706_145429](https://user-images.githubusercontent.com/3944034/86630578-043a7f00-bf9b-11ea-9d5e-3e1baf84c8a6.png)
- But I also noticed on this host (and others) this happening regularly (quite often in fact):
```
SELinux: mount invalid.  Same superblock, different security settings for (dev mqueue, type mqueue)
```

**Comment by guimou** [comment]
Further info:
- The problem now also appears on the "staging" namespace.
- There was an automated cluster update yesterday (OpenShift Dedicated environment). OCP now at 4.3.25. Problem persists.
- The update was stalled at some point (node update) because one of the failed pod was stuck in "Terminating" state for a few hours, which prevented node drain. Don't know if it's related though. After manually force-deleting the pod the process could continue.

**Comment by joscollin** [comment]
@guimou 
I just had a quick look at this issue. Is the monitor reachable from the csi-cephfsplugin container?

**Comment by guimou** [comment]
@joscollin Sorry, did not see the notification for your comment.
Yes, it does. And in fact all the rook-ceph install had been working perfectly for some time. It's only when we deployed those automated pipelines that runs multiple containers simultaneously with multiple mounts inside that we began to experience the issue.

**Comment by guimou** [comment]
More details on what works and what does not.
First, the problem is not linked to a specific container image, I was able to reproduce with other ones.

By replicating I was able to follow this trace of events:
- During container creation:
`Successfully assigned staging/test-ztph6 to ip-10-0-134-186.ca-central-1.compute.internal`
then
`AttachVolume.Attach succeeded for volume "pvc-1b42e048-db99-4db1-858d-b208585f4c3c"`
So it appeared that the volume was attached.
- But then: 
`Unable to attach or mount volumes: unmounted volumes=[hmf-data-restricted-raw-ccso], unattached volumes=[hmf-data-public hmf-data-restricted-raw-ccso hmf-data-restricted-processed-ccso hmf-data-restricted-raw-moh hmf-data-restricted-processed-moh default-token-vhnwz hmf-data-confidential]: timed out waiting for the condition`
- So I checked this particular volume that would not mount, **hmf-data-restricted-raw-ccso**
PVC hmf-data-restricted-raw-ccso => PV pvc-1b42e048-db99-4db1-858d-b208585f4c3c
- I connected directly as root to the host where the container was scheduled:
```
sh-4.4# mount | grep pvc-1b42
172.30.151.123:6789,172.30.109.18:6789,172.30.154.247:6789:/volumes/csi/csi-vol-cba856b8-bd42-11ea-91b6-0a580a800279/ad6e4442-5415-4837-bcce-8c8c232b93fb on /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-1b42e048-db99-4db1-858d-b208585f4c3c/globalmount type ceph (rw,relatime,name=csi-cephfs-node,secret=<hidden>,acl,mds_namespace=myfs)
```
So it appeared that the volume was indeed mounted.
- Let's check this:
```
sh-4.4# cd /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-1b42e048-db99-4db1-858d-b208585f4c3c/globalmount

sh-4.4# ls -la
total 0
drwxrwxrwx  4 root       root  2 Jul  3 16:13 .
drwxr-x---. 3 root       root 46 Jul  8 14:59 ..
drwxr-xr-x  2 1002590000 root  0 Jul  3 16:13 .ipynb_checkpoints
drwxr-xr-x  2 1002590000 root  6 Jul  3 16:13 ccis
```
Which is the content of the volume that I tried to mound in the container.

So apparently there is no issue with the storage, at least that I can identify. The volume is attached and mounted on the host, it just won't mount in the container.
Any idea where I should look next?
